src/HttpHeader.cc

```diff
@@ -1500,6 +1498,13 @@ HttpHeaderEntry::HttpHeaderEntry(Http::HdrType anId, const SBuf &aName, const ch
     if (id != Http::HdrType::BAD_HDR)
         ++ headerStatsTable[id].aliveCount;
 
+    if (id == Http::HdrType::CONNECTION)
+        alternateId = Http::HdrType::PROXY_CONNECTION;
+    else if (id == Http::HdrType::PROXY_CONNECTION)
+        alternateId = Http::HdrType::CONNECTION;
+    else // TODO: add other alternate ids here
+        alternateId = id;
```

Comment:
Please move this code into the sortInto() loop, making alternateId a local constant variable inside that loop and inlining hasId() changes. Something along these lines may work there:

```C++
/// helps treat Connection and Proxy-Connection positions the same way
const auto alternateId =
    (id == Http::HdrType::CONNECTION) ? Http::HdrType::PROXY_CONNECTION:
    (id == Http::HdrType::PROXY_CONNECTION) ? Http::HdrType::CONNECTION:
    id;
```

Reply:
Done.